### PR TITLE
Restructured the integrations chapter

### PR DESCRIPTION
**Pull request by anton-bobkov**
### Changelog category 

* Documentation (changelog entry is not required)

### Additional information

Restructured the Integrations chapter. Added subfolders for each category of articles.

### CODE REVIEW

File: ydb/docs/en/core/integrations/ingestion/toc-ingestion.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+items:
+- name: Log records collection using FluentBit
+  href: fluent-bit.md
+- name: Logstash plugins
+  href: logstash.md
+- name: Importing from JDBC data sources
+  href: import-jdbc.md
+- name: Spring Data JDBC
+  href: spring-data-jdbc.md
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/ingestion/toc-ingestion.yaml:9]: Afaik it's a more generic library than just for injection: let's create a new category called https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping out of Spring Data and Hibernate

File: ydb/docs/en/core/integrations/migration/_includes/toc-table.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Data Migrations
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/migration/_includes/toc-table.md:1]: Currently, most of the headers in the documentation capitalize only the first letter; let's keep it consistent.

I'd personally also prefer to switch over to some variation of headline case https://en.wikipedia.org/wiki/Title_case at some point, but let's discuss and potentially do this separately.

File: ydb/docs/en/core/integrations/ingestion/index.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Data Ingestion {#ingestion}
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/ingestion/index.md:1]: I don't immediately see the reason for it, but this `{#ingestion}` is getting rendered when it shouldn't:
Screenshot 2024-09-09 at 11 51 56

Maybe anchors are not supposed to be on `<h1>` /  `#`?

File: ydb/docs/en/core/integrations/toc_i.yaml [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
-- name: Importing from JDBC data sources
-  href: import-jdbc.md
-- name: DBeaver
-  href: ide/dbeaver.md
-- name: DataGrip
-  href: ide/datagrip.md
-- name: Spring Data JDBC
-  href: spring-data-jdbc.md
+- name: Graphical Interface Clients
+  href: gui_clients/index.md
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/toc_i.yaml:3]: 1. "GUI" is a well-known abbreviation; let's eliminate the "Clients" and `_clients` parts.
2. It won't be relevant after the previous item, but at the moment, the documentation is in the middle of refactoring from `foo_bar_baz` to `foo-bar-baz` in the URLs, and thus filenames.

```suggestion
- name: Graphical User Interfaces
  href: gui/index.md
```

File: ydb/docs/en/core/integrations/migration/toc-migration.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+items:
+- name: Schema migrations with goose
+  href: goose.md
+- name: Schema migrations with Liquibase
+  href: liquibase.md
+- name: Schema migrations with Flyway
+  href: flyway.md
+- name: Using Hibernate
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/migration/toc-migration.yaml:8]:
```suggestion
- name: Hibernate
```

File: ydb/docs/en/core/integrations/migration/toc-migration.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+items:
+- name: Schema migrations with goose
+  href: goose.md
+- name: Schema migrations with Liquibase
+  href: liquibase.md
+- name: Schema migrations with Flyway
+  href: flyway.md
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/migration/toc-migration.yaml:7]:
```suggestion
- name: Flyway
  href: flyway.md
- name: goose
  href: goose.md
- name: Liquibase
  href: liquibase.md
```

**Review by blinkov — DISMISSED** [review]

File: ydb/docs/en/core/integrations/gui/_includes/toc-table.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Graphical user interfaces
+
+|  Environment | Instruction | Compatibility Level |
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/gui/_includes/toc-table.md:3]:
```suggestion
|  Environment | Instruction | Compatibility level |
```

**Comment by blinkov** [comment]
lgtm

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**